src/components/TransactionConfirmationModal/index.tsx

```diff
@@ -173,16 +173,27 @@ export function ConfirmationModalContent({
   bottomContent,
   onDismiss,
   topContent,
+  chainId,
 }: {
   title: ReactNode
   onDismiss: () => void
   topContent: () => ReactNode
   bottomContent?: () => ReactNode | undefined
+  chainId?: SupportedL2ChainId | undefined
 }) {
+  const info = getChainInfo(chainId)
```

Comment:
Why not just get `chainId` here and check if its L2? I think encapsulating that logic within this component makes a bit more sense because:
- we can check whether it renders a badge for L2 or not (which is good)
- it is less confusing - undefined `chainId` means to me that we may not be connected to any chain and it's not clearly visible from the following code that we want to render for L2 only

Reply:
Okay, I see, this may not work because of the fact we need modalState to decide whether we want to show or not. In that case, I would recommend you stick to `headerContent` pattern (function to render modal elements) and we just repeat that.

I would add a TODO to Linear that we need this to be refactored similarly how you refactored in your last PR from array/dynamic rendering to static.

Ideally, there should be a simple if conditional / switch statement, three different modals for each stage and each one having all the elements explicitly defined. 

If you have capacity this week, you can pick this work up. If not, feel free to assign it to me and I will work on it this week.